src/input/input.directive.ts

```diff
@@ -38,4 +38,20 @@ export class TextInput {
 	@HostBinding("class.bx--text-input--light") get isLightTheme() {
 		return this.theme === "light";
 	}
+	@HostBinding("attr.readonly") readonlyAttr = null;
+	@Input()
+	public set readonly(value) {
+		this._readonly = value;
+		// Set attribute based on input
+		if (this._readonly === true) {
+			this.readonlyAttr = "true";
+		} else if (this._readonly === false) {
+			this.readonlyAttr = null;
+		}
```

Comment:
I'm not sure about this... because the native `readonly` attribute would act as if it was `true` if it was set to some random string, for example. If we make our `readonly` act as if it's `false` when getting a random string as input, then that would be even more confusing, hahah...

If I were using this on the other end, I think I'd want to see an error being thrown if I'm passing in the wrong type of thing. Typically, typescript should handle this, but I understand if not everybody has it fully working, especially for Carbon.

Reply:
Actually, never mind... because the `ibm-label` input for `readonly` works like `==` rather than `===` to apply the style:
`@HostBinding("class.bx--text-input-wrapper--readonly") readonly`